processor/resourcedetectionprocessor/internal/aws/elasticbeanstalk/elasticbeanstalk.go

```diff
@@ -0,0 +1,83 @@
+// Copyright -c OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticbeanstalk
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strconv"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	TypeStr     = "elasticBeanstalk"
+	linuxPath   = "/var/elasticbeanstalk/xray/environment.conf"
+	windowsPath = "C:\\Program Files\\Amazon\\XRay\\environment.conf"
+)
+
+var _ internal.Detector = (*Detector)(nil)
+
+type Detector struct {
+	fs readOnlyFileSystem
+}
+
+type EbMetaData struct {
+	DeploymentID    int    `json:"deployment_id"`
+	EnvironmentName string `json:"environment_name"`
+	VersionLabel    string `json:"version_label"`
+}
+
+func NewDetector() (internal.Detector, error) {
+	return &Detector{fs: &ebReadOnlyFileSystem{}}, nil
+}
+
+func (d Detector) Detect(context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	var conf file
+	var err error
+
+	if d.fs.IsWindows() {
+		conf, err = d.fs.Open(windowsPath)
+	} else {
+		conf, err = d.fs.Open(linuxPath)
+	}
+
+	if err != nil {
+		log.Println("Unable to find Elastic Beanstalk config file, skipping resource detection")
```

Comment:
Shouldn't be using log package but the zap logger passed into the component (looks like ecs is using log too which it shouldn't have). We're not passing log instances into detectors right now so will require some refactoring to add it.

Reply:
Yeah I noticed there's no Zap object passed in which is why I fell back to `log`, but in retrospect I probably should have just done nothing. I can remove the logging statements from this PR, and take a look at the refactor to pass in the zap logger in a separate PR.